metadoc-js/src/main/scala/metadoc/ScalaDocumentSymbolProvider.scala

```diff
@@ -0,0 +1,53 @@
+package metadoc
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation._
+import scala.meta.{Attributes, Denotation}
+import metadoc.schema.{Index, Position, Symbol}
+import monaco.CancellationToken
+import monaco.editor.IReadOnlyModel
+import monaco.languages.{DocumentSymbolProvider, SymbolInformation, SymbolKind}
+
+@ScalaJSDefined
+class ScalaDocumentSymbolProvider(attrs: Attributes)
+    extends DocumentSymbolProvider {
+  override def provideDocumentSymbols(
+      model: IReadOnlyModel,
+      token: CancellationToken
+  ) = {
+    val denotations = attrs.denotations.map { case (s, d) => s -> d }.toMap
+    val symbols = for {
+      (symPos, sym) <- attrs.names
+      denotation <- denotations.get(sym)
+      kind <- symbolKind(denotation)
+    } yield {
+      val pos = Position("???", symPos.start.offset, symPos.end.offset)
+      new SymbolInformation(
+        name = denotation.name,
+        containerName = "???",
```

Comment:
The fully qualified names make sense to me especially for cross project metadoc sites.